src/MagnumPlugins/WebPImporter/WebPImporter.cpp

```diff
@@ -0,0 +1,168 @@
+    /* Because here we're copying the data and using the _in to check if file
+       is opened, having them nullptr would mean openData() would fail without
+       any error message. It's not possible to do this check on the importer
+       side, because empty file is valid in some formats (OBJ or glTF). We also
+       can't do the full import here because then doImage2D() would need to
+       copy the imported data instead anyway (and the uncompressed size is much
+       larger). This way it'll also work nicely with a future openMemory(). */
+    if(data.isEmpty()) {
+        Error{} << "Trade::WebpImporter::openData(): the file is empty";
+        return;
+    }
+
+    /* Take over the existing array or copy the data if we can't */
+    if(dataFlags & (DataFlag::Owned|DataFlag::ExternallyOwned)) {
+        _in = std::move(data);
+    } else {
+        _in = Containers::Array<char>{NoInit, data.size()};
+        Utility::copy(data, _in);
+    }
+}
+
+UnsignedInt WebPImporter::doImage2DCount() const { return 1; }
+
+Containers::Optional<ImageData2D> WebPImporter::doImage2D(UnsignedInt, UnsignedInt) {
+
+    /* Create the structure that contains the WebP image data. */
+    WebPData webp_data;
+    webp_data.bytes = reinterpret_cast<uint8_t*>(_in.data());
+    webp_data.size = _in.size();
+
+    /* Verify the file is a WebP image file */
+    bool isWebPImage = WebPGetInfo(webp_data.bytes, webp_data.size, nullptr, nullptr);
+    if(!isWebPImage) {
+        Error() << "Trade::WebpImporter::image2D(): wrong file format";
+        return Containers::NullOpt;
+    }
+
+    /* Reading the file information */
+    //WebPBitstreamFeatures info;
+    //CORRADE_ASSERT(WebPGetFeatures(webp_data.bytes, webp_data.size, &info) == VP8_STATUS_OK, "The features couldn't be retrieved.", Containers::NullOpt);
+
+    WebPDecoderConfig config;
+    CORRADE_ASSERT(WebPInitDecoderConfig(&config), "Decoder version mismatch", Containers::NullOpt);
+    CORRADE_ASSERT(WebPGetFeatures(webp_data.bytes, webp_data.size, &config.input) == VP8_STATUS_OK, "The features couldn't be retrieved.", Containers::NullOpt);
+
+
+    /* Image size, channels and pixel format */
+    const Vector2i size(config.input.width, config.input.height);
+
+    int channels = 3;
+    PixelFormat pixel_format = PixelFormat::RGB8Unorm;
+    WEBP_CSP_MODE colour_depth = MODE_RGB;
+
+    if(config.input.has_alpha) {
+        channels = 4;
+        pixel_format = PixelFormat::RGBA8Unorm;
+        colour_depth = MODE_RGBA;
+    }
+
+    /* decompression of the image */
+    WebPDecBuffer* const output = &config.output;
+    output->colorspace = colour_depth;
+    output->u.RGBA.size = size.x() * size.y() * channels;
+    Containers::Array<char> outData{NoInit, output->u.RGBA.size};
+    auto* rgba_buf = reinterpret_cast<uint8_t*>(outData.data());
+    output->u.RGBA.rgba = rgba_buf;
+    output->u.RGBA.stride = size.x() * channels;
```

Comment:
`ImageData` expects row stride to be divisible by 4 by default (carried over from OpenGL, it's because such images have faster memory access). One option would be to pass a `PixelStorage{}.setAlignment(1)` to the `ImageData` constructor if stride isn't divisible by 4, but since WebP has an option to configure the stride, I think a better option would be to just make the rows four-byte aligned always:

```cpp
/* Round up to a multiple of 4 bytes */
const std::size_t stride = 4*((size.x()*channels + 3)/4); 
output->u.RGBA.size = stride*size.y();
output->u.RGBA.stride = stride;
...
```

When you get to testing, be sure to add a dedicated test with an image that has the stride not divisible by 4 (so, say, a 7x3 RGB image).